lib/api/expect/cookie.js

```diff
@@ -48,11 +48,18 @@ class ExpectCookie extends BaseExpect {
     promise
       .then((result) => {
         const domainStr = this.cookieDomain ? ` for domain "${this.cookieDomain}"` : '';
-
         if (!result.value) {
           return this.reject(new NotFoundError(`no cookie "${this.cookieName}"${domainStr} was found`));
         }
-        this.resultValue = result ? result.value : {};
+        if (result && result.value) {
+          if (Array.isArray(result.value)) {
```

Comment:
Can you explain why this change is necessary?

Reply:
Selenium is returning an object now with the entire cookie contents(e.g. is it secure, domain, etc) not just the value of the cookie. The API is only expecting the value in the tests so keeps things as they want.

Reply:
But that's for Selenium 4. Chromedriver response remains unchanged, isn't that the case? In which case, this changes seems quite risky and it it's not clear for me if the old format is still going to work.

Reply:
Chromedriver is only returning what the [W3C standard](https://w3c.github.io/webdriver/#dfn-serialized-cookie) says. 

We can change it to return the value, which is the way the current API is, and codified above, or we can change expect to compare objects which the old code is going to need to do but will break folks tests if they upgrade.

What do you want to do here? I think the least risky approach is what I have done here.

Reply:
I don't think we need to handle both cases. Speaking to other Selenium developers is that Chrome JWP version has had this for a very long time. See conversation from on [slack](https://seleniumhq.slack.com/archives/CBH302726/p1645548272153689)/[matrix](https://matrix.to/#/!ILwPlTupoMrQwfDHYj:matrix.org/$J8eXAFPPA74iEhwLhP5aiFJEoRHDN-TiCEpCzkMYMQE?)